econml/validate/drtester.py

```diff
@@ -0,0 +1,796 @@
+        if (self.cate_preds_val is None) or (reg_cate is not None):
+            if (Zval is None) or (reg_cate is None):  # need at least Zval and a CATE estimator to fit
+                raise Exception('CATE not yet fitted - must provide Zval and CATE estimator')
+            self.fit_cate(reg_cate, Zval, Ztrain)
+
+        if self.n_treat == 1:  # binary treatment
+            reg = OLS(self.dr_val, add_constant(self.cate_preds_val)).fit()
+            params = [reg.params[1]]
+            errs = [reg.bse[1]]
+            pvals = [reg.pvalues[1]]
+        else:  # categorical treatment
+            params = []
+            errs = []
+            pvals = []
+            for k in range(self.n_treat):  # run a separate regression for each
+                reg = OLS(self.dr_val[:, k], add_constant(self.cate_preds_val[:, k])).fit(cov_type = 'HC1')
+                params.append(reg.params[1])
+                errs.append(reg.bse[1])
+                pvals.append(reg.pvalues[1])
+
+        self.blp_res = pd.DataFrame(
+            {'treatment': self.tmts[1:], 'blp_est': params, 'blp_se': errs, 'blp_pval': pvals}
+        ).round(3)
+
+        return self
+
+    def evaluate_all(
+        self,
+        reg_cate=None,
+        Zval: np.array = None,
+        Ztrain: np.array = None,
+        n_groups: int = 4
+    ):
+
+        """
+        Implements the best linear prediction (`evaluate_blp'), calibration (`evaluate_cal') and QINI coefficient
+        (`evaluate_qini') methods.
+
+        Parameters
+        ----------
+        reg_cate: estimator, default ``None''
+            CATE model. Must be able to implement `fit' and `predict' methods. If not specified, then fit_cate() must
+            already have been implemented
+        Zval: (n_cal x k) matrix, default ``None''
+            Validation sample features for CATE model. If not specified, then `fit_cate' method must already have been
+            implemented
+        Ztrain: (n_train x k) matrix, default ``None''
+            Training sample features for CATE model. If not specified, then `fit_cate' method must already have been
+            implemented
+
+        Returns
+        -------
+        self, with added dataframe df_res summarizing the results of all tests
+        """
+
+        # if CATE is given explicitly or has not been fitted at all previously, fit it now
+        if (self.cate_preds_val is None) or (self.cate_preds_train is None) or (reg_cate is not None):
+            if (Zval is None) or (reg_cate is None) or (Ztrain is None):
+                raise Exception('CATE not fitted on training data - must provide CATE model and both Zval, Ztrain')
+            self.fit_cate(reg_cate, Zval, Ztrain)
+
+        self.evaluate_blp()
+        self.evaluate_cal(n_groups=n_groups)
+        self.evaluate_qini()
+
+        self.df_res = self.blp_res.merge(self.qini_res, on='treatment')
+        self.df_res['cal_r_squared'] = np.around(self.cal_r_squared, 3)
+
+        return self
```

Comment:
Maybe here you can return an EvaluationResults object that has each of them as a property, i.e. 
res = evaluate_all(cate)
res.blp.summary()
res.cal.summary()
res.qini.summary()
res.cal.plot_cal(...)
res.summary() # plots all summary results all together